pandas/core/internals/concat.py

```diff
@@ -449,6 +414,40 @@ def get_reindexed_values(self, empty_dtype: DtypeObj, upcasted_na) -> ArrayLike:
         return values
 
 
+def make_na_array(dtype: DtypeObj, shape: Shape) -> ArrayLike:
```

Comment:
cool, FYI https://github.com/pandas-dev/pandas/blob/master/pandas/core/dtypes/cast.py#L1850 are somewhat similar in function (though not na)